### PR TITLE
Adding ReleaseManagement auto generated node js client.

### PR DESCRIPTION
**Pull request by ankitgo**

### CODE REVIEW

File: api/JenkinsApi.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+* ---------------------------------------------------------
+* Copyright(C) Microsoft Corporation. All rights reserved.
+* ---------------------------------------------------------
+*/
+
+// Licensed under the MIT license.  See LICENSE file in the project root for full license information.
+
+/// <reference path="../node/node.d.ts"/>
+/// <reference path="../q/Q.d.ts"/>
+
```
Comment (bryanmacfarlane) [inline api/JenkinsApi.ts]: I'm struggling with JenkinsApi being in the agent api.  it's not part of the agent or it's api.  This seems like an internal implementation detail of your RM plugin

File: api/ReleaseManagementApi.ts [hunk 2]
```diff
@@ -0,0 +1,1747 @@
+/*
```
Comment (bryanmacfarlane) [inline api/ReleaseManagementApi.ts]: This is all generated so I'm assuming it's good and you tested

File: api/WebApi.ts [hunk 3]
```diff
@@ -139,4 +141,16 @@ export class WebApi {
...
+
+    public getReleaseManagemntApi(serverUrl: string = this.serverUrl, handlers: VsoBaseInterfaces.IRequestHandler[] = [this.authHandler]): releasemanagementm.IReleaseManagementApi {
+        return new releasemanagementm.ReleaseManagementApi(serverUrl, handlers);
+    }
+
+    public getQReleaseManagemntApi(serverUrl: string = this.serverUrl, handlers: VsoBaseInterfaces.IRequestHandler[] = [this.authHandler]): releasemanagementm.IQReleaseManagementApi {
+        return new releasemanagementm.QReleaseManagementApi(serverUrl, handlers);
+    }
+
+    public getJenkinsApi(serverUrl: string = this.serverUrl, handlers: VsoBaseInterfaces.IRequestHandler[] = [this.authHandler]): jenkinsm.IJenkinsApi {
```
Comment (bryanmacfarlane) [inline api/WebApi.ts]: Shouldn't be here.

**Comment by bryanmacfarlane** [comment]
Reviewed changes.  (1) React to conflicts (2) JenkinsApi should not be here - move to internal implementation of your agent plugin.

File: api/ReleaseManagementApi.ts (hunk 2 above)
Comment (ankitgo) [inline api/ReleaseManagementApi.ts]: Yes

File: api/WebApi.ts (hunk 3 above)
Comment (ankitgo) [inline api/WebApi.ts]: only Jenkins ?

**Pull request closed without merging**